main.py

```diff
@@ -51,13 +68,16 @@ def __init__(self):
 
     @endpoints.method(
         OAUTH_CODE_RESOURCE,
-        LinkInfoResponse,
+        AccessTokenResponse,
         path='fence/oauthcode',
         http_method='POST',
         name='fence/oauthcode')
     def oauthcode(self, request):
         user_info = self.auth.require_user_info(self.request_state)
-        return LinkInfoResponse(expires=datetime.now(), username=request.oauthcode)
```

Comment:
I think this should return LinkInfoResponse. Do we get any of this information from fence?